### PR TITLE
Use media library to select placeholder image

### PR DESCRIPTION
**Pull request by ruscoe**
This pull request introduces a media library button / preview image to the Customizer / Misc settings tab.

screen shot 2018-09-01 at 2 41 20 pm

screen shot 2018-09-01 at 2 41 36 pm

screen shot 2018-09-01 at 2 41 59 pm

### CODE REVIEW

File: woocommerce-customizer.php [hunk 1]
```diff
@@ -334,6 +337,41 @@ public static function render_outdated_wc_version_notice() {
...
+		 * Get the placeholder image URL directly from settings so the admin form
+		 * shows the correct image immediately after saving.
+		 */
+		$placeholder_url = $this->settings->customizations["woocommerce_placeholder_img_src"];
+
+		?>
+		<table class="form-table">
+			<tbody><tr valign="top">
+				<th scope="row" class="titledesc">
+					<label for="woocommerce_placeholder_img_src"><?php _e( 'Placeholder Image' ); ?></label>
```
Comment (bekarice) [inline woocommerce-customizer.php]: We tend to escape translations given we're not often working directly with translators we trust. Also need a textdomain here for localization. So we'd use `esc_html_e( 'Placeholder image', 'woocommerce-customizer' );`

File: woocommerce-customizer.php [hunk 2]
```diff
@@ -334,6 +337,41 @@ public static function render_outdated_wc_version_notice() {
...
+				</th>
+				<td class="forminp forminp-text">
+					<div class="media-library-wrapper">
+						<img id="media_preview" src="<?php echo $placeholder_url; ?>" style="max-height: 200px;">
```
Comment (bekarice) [inline woocommerce-customizer.php]: use `esc_url()`

File: woocommerce-customizer.php [hunk 3]
```diff
@@ -334,6 +337,41 @@ public static function render_outdated_wc_version_notice() {
...
+					</div>
+					<input id="media_library_button" type="button" class="button" value="<?php _e( 'Choose image' ); ?>" />
```
Comment (bekarice) [inline woocommerce-customizer.php]: Same comments RE escaping translations / textdomain. However, in this case, `esc_attr_e( 'string', 'textdomain' );` is more appropriate

File: woocommerce-customizer.php [hunk 4]
```diff
@@ -334,6 +337,41 @@ public static function render_outdated_wc_version_notice() {
...
+					<input type="hidden" name="woocommerce_placeholder_img_src" id="woocommerce_placeholder_img_src" value="<?php echo $placeholder_url; ?>">
```
Comment (bekarice) [inline woocommerce-customizer.php]: use `esc_url()`

**Comment by ruscoe** [comment]
Thanks for the review and awesome notes, @bekarice! I'll get these changes in ASAP.

**Comment by ruscoe** [comment]
@bekarice I think this is ready for another pass, whenever you have time :+1: 

Thanks again for the feedback.

**Pull request closed without merging**